hadoop/src/main/java/org/apache/carbondata/hadoop/util/CarbonInputFormatUtil.java

```diff
@@ -70,6 +69,18 @@ public static CarbonQueryPlan createQueryPlan(CarbonTable carbonTable, String co
           i++;
         }
       }
+    } else {
+      // query all columns
+      List<CarbonMeasure> tableMsrs = carbonTable.getMeasureByTableName(factTableName);
```

Comment:
Because when I run HadoopFileExample, it is returning empty row. Then I found out that it is because projection columns are not set in `CarbonInputFormat`. So I have added it back. If it is removed, then it is mandatory that user set projection columns in `CarbonInputFormat`